### DISCUSSION TITLE
XAUTH Vici

### DISCUSSION THREAD

**Discussion by YafimK** [strongswan/strongswan#1518] [Q&A]
Hey,
I'm exploring authentication alternatives for radius, where I can use some local daemon with access to DB to approve sessions.
I've looked at the different plugins, including ext-auth, but that doesn't offer verifying passwords passed from the user.
The other option is using the SQL plugin, but it needs to be marked as stable (?).
Currently, the "basic" method is getting events such as "SA UP" checking its params (but without password) and dropping it if not approved. 

Can the VICI protocol be used for it somehow? Do you think it's acceptable by design if it's developed as PR?

**Comment by tobiasbrunner** [comment]
> where I can use some local daemon with access to DB to approve sessions.

"Approve" i.e. authorize sessions is what you can do with the [_ext-auth_ plugin](https://docs.strongswan.org/docs/5.9/plugins/ext-auth.html) (although, only based on selected information). But I guess you mean something else and actually want to authenticate the clients too. Also note that you could easily run a RADIUS server as "local daemon" for both.

> but that doesn't offer verifying passwords passed from the user.

Note that only IKEv1 XAUTH and IKEv2 EAP GTC (which not many clients support) will send the client's plaintext password to the server. So depending on how your DB stores the passwords (e.g. hashed in some way), verifying them might not be possible (e.g. with the challenge-response-based EAP-MD5 or EAP-MSCHAPv2 the server needs the plaintext passwords stored).

> The other option is using the SQL plugin, but it needs to be marked as stable (?).

What do you mean with "marked as stable"?

> Currently, the "basic" method is getting events such as "SA UP" checking its params (but without password) and dropping it if not approved.

That's what the _ext-auth_ plugin offers (unless you need some information that's not provided by the plugin).

> Can the VICI protocol be used for it somehow?

For what exactly? You could definitely use it to load secrets from some third-party source (not on-demand, though).
